File: MvpJava/src/main/java/com/base/applemvp/common/BaseDialog.java
/**
 * 公共父类弹出框
 */
package com.base.applemvp.common;

import android.app.Activity;
import android.app.Dialog;
import android.content.Context;
import android.content.res.Resources;
import android.util.DisplayMetrics;
import android.view.View;
import android.view.ViewGroup.LayoutParams;


/**
 * @author applehsp
 */
public abstract class BaseDialog extends Dialog  {

	public Activity  activity;

	public  View contentView;

	public BaseDialog(Activity mActivity, int style) {
		super(mActivity, style);
		activity = mActivity;
		contentView=getLayoutInflater().inflate(layoutId(),null);
		setContentView(contentView);
		initView();
		initData();
	}



	@Override
	public void dismiss() {
		super.dismiss();
	}



	private float getDensity(Context context) {
		Resources resources = context.getResources();
		DisplayMetrics dm = resources.getDisplayMetrics();
		return dm.density;
	}


	protected abstract int layoutId();

	protected abstract void initView();

	protected abstract void initData();

	/**
	 * 显示对话框
	 * */
	public void goShow() {
		this.setCanceledOnTouchOutside(true);
		getWindow().setLayout(LayoutParams.WRAP_CONTENT,
				LayoutParams.WRAP_CONTENT);
		show();
	}






}

File: BaseMvp/src/main/java/com/base/applemvp/common/BaseDialog.kt
/**
 * 公共父类弹出框
 */
package com.base.applemvp.common

import android.app.Activity
import android.app.Dialog
import android.content.Context
import android.view.View
import android.view.ViewGroup

/**
 * @author applehsp
 */
abstract class BaseDialog(var activity: Activity, style: Int) : Dialog(activity, style) {

    internal  var contentView: View

    override fun dismiss() {
        super.dismiss()
    }

    private fun getDensity(context: Context): Float {
        val resources = context.resources
        val dm = resources.displayMetrics
        return dm.density
    }

    protected abstract fun layoutId(): Int
    protected abstract fun initView()
    protected abstract fun initData()
    /**
     * 显示对话框
     */
    fun goShow() {
        setCanceledOnTouchOutside(true)
        window.setLayout(ViewGroup.LayoutParams.WRAP_CONTENT,
                ViewGroup.LayoutParams.WRAP_CONTENT)
        show()
    }

    init {
        contentView = layoutInflater.inflate(layoutId(), null)
        setContentView(contentView)
        initView()
        initData()
    }
}